### PR TITLE
Refactored fetch calls

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**
All services are called in clientId-page.

### CODE REVIEW

File: app/javascript/Clients/clientIdPage/ClientIdPage.js [hunk 1]
```diff
@@ -82,17 +127,58 @@ class ClientIdPage extends React.Component {
...
           },
         });
       })
       .catch(() => this.setState({ relatedClients: { XHRStatus: 'error' } }));
   };
 
-  handleSelect(href, event) {
-    event.stopPropagation();
-  }
+  handleSelect = (href, event) => event.stopPropagation();
```
Comment (kelfink) [inline app/javascript/Clients/clientIdPage/ClientIdPage.js]: Redeclaring this method?  Looks like the same logic to stop the event propagation

Comment (Sushmareddy-OSI) [reply app/javascript/Clients/clientIdPage/ClientIdPage.js]: the other method is removed in commit 6171d38d3d9a1771385588cac7fcc8b093516231

File: app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+export default class ClientInformation extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      childClient: props.childClient,
+      checked: false,
+      value: '',
+      client: ['Client is a Minor/NMD parent'],
+      clients: [' Client is a Safely surrendered baby'],
```
Comment (kelfink) [inline app/javascript/Clients/clientIdPage/ClientInformation/ClientInformation.js]: space before string is odd.  Is this verbatim test from someplace?  Seems like capitalization should be changed:  suggest:  "Client is a safely-surrendered baby" unless Safely is a recognized term.

File: app/javascript/Clients/clientIdPage/ClientInformation/Sogie.js [hunk 3]
```diff
@@ -48,68 +40,72 @@ export default class Sogie extends React.Component {
...
+              <TextArea
+                name="Description"
+                rows={10}
+                label="Description"
+                value={sexual_description}
+                handleOnChange={this.handleOnChange('sexual_description')}
+              />
+            </div>
+          )}
+          {sexual_orintation ===
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/ClientInformation/Sogie.js]: sexual_orientation

File: app/javascript/Clients/clientIdPage/Relationships/RelationsCard.test.js [hunk 4]
```diff
@@ -5,17 +5,49 @@ import RelationsCard from './RelationsCard.js';
...
     }).not.toThrow();
   });
 
   it('renders BootsstrapTable Component', () => {
     expect(component.find('BootstrapTable').length).toEqual(1);
     expect(component.find('TableHeaderColumn').length).toEqual(7);
   });
+
+  it('formats the props been passed from the parent component', () => {
+    const props = [
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/Relationships/RelationsCard.test.js]: No need to test the formatter here it's already tested in the formatter tests.

File: app/javascript/Clients/clientIdPage/ClientInformation/OtherClientInformation.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    let childClient = nextProps.childClient;
+    this.setState({ childClient });
+  }
+
+  handleDropdownChange = name => ({ value }) => {
+    const { childClient } = this.state;
```
[the remark this answers is not in the document]
Comment (Sushmareddy-OSI) [reply app/javascript/Clients/clientIdPage/ClientInformation/OtherClientInformation.js]: Using this handleDropdownChange for multiple dropdown's by passing the value.

File: app/javascript/Clients/clientIdPage/Relationships/RelationsCard.test.js (hunk 4 above)
Comment (mlipardo) [reply app/javascript/Clients/clientIdPage/Relationships/RelationsCard.test.js]: I think the description is misleading. I will change the description. This test is about data passing from the parent component and it is rendering in the bootstrap-table and it has props.

**Pull request merged**